rosbag2_transport/test/rosbag2_transport/test_keyboard_controls.cpp

```diff
@@ -203,13 +203,12 @@ TEST_F(RecordIntegrationTestFixture, test_keyboard_controls)
 
   recorder->record();
 
-  this->start_async_spin(recorder);
+  start_async_spin(recorder);
+  auto cleanup_process_handle = rcpputils::make_scope_exit([&]() {stop_spinning();});
 
   EXPECT_THAT(recorder->is_paused(), true);
   keyboard_handler->simulate_key_press(rosbag2_transport::Recorder::kPauseResumeToggleKey);
   EXPECT_THAT(recorder->is_paused(), false);
   keyboard_handler->simulate_key_press(rosbag2_transport::Recorder::kPauseResumeToggleKey);
   EXPECT_THAT(recorder->is_paused(), true);
-
-  this->stop_spinning();
```

Comment:
@clalancette I don't know how, but it seems we got a regression in test_keyboard_controls.cpp file. It fails on RPR job and on Windows CI. I've never seen such a failure before.
Thoughts?

Reply:
Yeah, I saw that.  Both in Rpr and in the Windows CI, those tests are timing out.  I can only imagine this is happening because we are never successfully `join()` on the spinning thread, but I can't reproduce locally at all (either on Linux or on Windows).  I'm going to have to poke at it more.